lib/OpenQA/WebSockets/Server.pm

```diff
@@ -202,36 +204,52 @@ sub _message {
 sub _get_dead_worker_jobs {
     my ($threshold) = @_;
 
+    my $schema = OpenQA::Schema::connect_db;
+
+    my $dtf = $schema->storage->datetime_parser;
+    my $dt = DateTime->from_epoch(epoch => time() - $threshold, time_zone => 'UTC');
+
     my %cond = (
-        state              => OpenQA::Schema::Result::Jobs::RUNNING,
-        'worker.t_updated' => {'<' => $threshold},
+        state              => [OpenQA::Schema::Result::Jobs::EXECUTION_STATES],
+        'worker.t_updated' => {'<' => $dtf->format_datetime($dt)},
     );
-    my %attrs = (join => 'worker',);
+    my %attrs = (join => 'worker', select => [qw/id state/]);
 
-    my $schema = OpenQA::Schema::connect_db;
     return $schema->resultset("Jobs")->search(\%cond, \%attrs);
 }
 
+sub _is_job_considered_dead {
+    my ($job) = @_;
+
+    # running jobs have the default timeout
+    if ($job->state eq OpenQA::Schema::Result::Jobs::RUNNING) {
+        return 1;
+    }
+    # much bigger timeout for uploading jobs
+    if ($job->state eq OpenQA::Schema::Result::Jobs::UPLOADING) {
+        my $delta = DateTime->now()->epoch() - $job->worker->t_updated->epoch();
+        return if $delta > 1000;
+    }
+    return 0;
```

Comment:
ah, right. I was thinking along the line, that a job in interactive should not be killed at all